app/models/character_attribute.rb

```diff
@@ -1,5 +1,7 @@
 class CharacterAttribute < ApplicationRecord
-  belongs_to :character_type
+  belongs_to :character_type, optional: true, inverse_of: :character_attributes
```

Comment:
@StefanGajic optional?

Reply:
I guess I was looking at older version of Rails again, now I see that in newer versions from late 2018 belongs_to associations are now automatically required: true